### DISCUSSION TITLE
Updating to MLKit 7

### DISCUSSION THREAD

**Discussion by bengavin** [AdamEssenmacher/GoogleApisForiOSComponents#62] [Ideas]
Good morning!

I'm in the process of putting together a PR to update the MLKit.BarcodeScanning dependencies to MLKit 7 [Sept 2024 release].  It looks like I'm going to need to update some of the 'core' libraries to versions beyond what Firebase 10.x supports.  Is there any plan to upgrade to the 11.5 release of Firebase, or was there a blocker discovered that prevents this before I start too far down this path? :)

Thanks!

**Comment by AdamEssenmacher** [comment]
I believe @Digifais had done some work towards Firebase 11, but I'm not sure where that is right now. Updating the Firebase suite over a major version is just a lot more effort than minor version bumps.

In the case of a minor version update, the effort needed to published a new set of bindings is usually just bumping dependency versions. It's not usually necessary to write bindings for any new native APIs that might have been exposed.

A major version bump is a signal that breaking changes have been made, so we need to be a lot more careful. Whole XCFramework dependencies might have been added or removed. Header files should be diffed to check for breaking API changes, and bindings need to be updated if any are found.

I recommend starting with 11.0 so that effort can be focused on the minimum changes needed to get through the major version, and then incrementally working up to 11.4. It's a bit easier to tackle these upgrades one bite at a time, and it's nice to have a published nuget for each version. Its rare, but sometimes bugs make it into the native libraries and it can be nice to have the option to downgrade to the last working version.

**Reply by bengavin** [reply]
Sounds good, I'll probably start from the MLKit side of the house and just upgrade the Firebase side until everything bundles and builds.  That should at least provide a starting point for upgrading the entirety of the Firebase support and shouldn't break anyone in the mean time.  I agree, jumping to 11.0 to start is a good approach, I just don't have any easy way to test any of the Firebase related code as I don't use it and I'm unsure of the state of the in-repo samples :)

**Reply by AdamEssenmacher** [reply]
> I just don't have any easy way to test any of the Firebase related code as I don't use it and I'm unsure of the state of the in-repo samples

Welcome to my world 😓 

The samples need to be overhauled or rewritten. My app uses a solid subset of the Firebase SDKs (but not all of them), but it certainly doesn't exercise 100% of the API surfaces. It's all I use to test, and unfortunately have to rely on community feedback otherwise.

**Reply by bengavin** [reply]
The PR here:

https://github.com/AdamEssenmacher/GoogleApisForiOSComponents/pull/63

Seems to work properly with an updated version of the BarcodeScanner.Mobile project [at least on the Maui side].  There were pretty minimal updates required on the Firebase side in the libraries I need to touch [Core, Installations].

**Reply by Digifais** [reply]
> I believe @Digifais had done some work towards Firebase 11, but I'm not sure where that is right now. Updating the Firebase suite over a major version is just a lot more effort than minor version bumps.
> 
> In the case of a minor version update, the effort needed to published a new set of bindings is usually just bumping dependency versions. It's not usually necessary to write bindings for any new native APIs that might have been exposed.
> 
> A major version bump is a signal that breaking changes have been made, so we need to be a lot more careful. Whole XCFramework dependencies might have been added or removed. Header files should be diffed to check for breaking API changes, and bindings need to be updated if any are found.
> 
> I recommend starting with 11.0 so that effort can be focused on the minimum changes needed to get through the major version, and then incrementally working up to 11.4. It's a bit easier to tackle these upgrades one bite at a time, and it's nice to have a published nuget for each version. Its rare, but sometimes bugs make it into the native libraries and it can be nice to have the option to downgrade to the last working version.

I unfortunately haven't had the time to put more work into Firebase 11 as of yet since it's very busy with the end of the year in sight at work...

**Comment by AdamEssenmacher** [comment]
@bengavin I just pushed nugets for MLKitBarcodeScanning v6. Please give them a try and let me know if they actually work 🙃
